app/utils/token.ts

```diff
@@ -76,8 +84,8 @@ export class Token {
 
 export const getNetwork = async (): Promise<INetwork> => {
   let network
-  if (new Token().has()) {
-    network = new Token().network
+  if (Token.getInstance().has()) {
+    network = Token.getInstance().network
   } else {
     network = await loadNetwork()
   }
```

Comment:
in general else blocks can be avoided (cyclomatic complexity mentioned by Justin C in code review)
```js
if (Token.getInstance().has()) {
  network = Token.getInstance().network
}

network = network || await loadNetwork()
```
It is not a request for change but is a good example for future code